### PR TITLE
[WIP] Ticket Purchase Index Page

### PR DESCRIPTION
**Pull request by siddhantbajaj**
screen shot 2017-06-26 at 6 09 35 am

### CODE REVIEW

File: app/controllers/ticket_purchases_controller.rb [hunk 1]
```diff
@@ -4,8 +4,15 @@ class TicketPurchasesController < ApplicationController
   authorize_resource :conference_registrations, class: Registration
 
   def create
+    if(params[:tickets].nil?)
```
Comment (Ana06) [inline app/controllers/ticket_purchases_controller.rb]: but in this action you want to create a ticket. What could you pay if there is nothing? :thinking:

**Comment by Ana06** [comment]
[Ticket purchase index page](https://trello.com/c/qrqc9KSz/10-ticket-purchase-index-page)

File: app/controllers/ticket_purchases_controller.rb (hunk 1 above)
Comment (siddhantbajaj) [inline app/controllers/ticket_purchases_controller.rb]: This is the case when ticket purchase is already created but unpaid. When the user click the pay button on ticket purchase index page, this method is called but there will be no params in this case as the ticket purchase is already created and we just need to redirect the user to new payment controller method.

File: app/views/ticket_purchases/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        Your unpaid ticket purchases for the conference
+
+    .col-md-12
+      - if @unpaid_ticket_purchases.present?
+        %table.table.table-bordered.table-striped.table-hover#roles
+          %thead
+            %th ID
+            %th Type
+            %th Quantity
+            %th User
```
Comment (siddhantbajaj) [inline app/views/ticket_purchases/index.html.haml]: Yes I'll make the change. 😉

File: app/views/conferences/_conference_details.html.haml [hunk 3]
```diff
@@ -34,6 +34,8 @@
               = link_to "My Proposals", conference_program_proposals_path(conference.short_title), class: 'btn btn-default'
             - elsif can? :new, conference.program.events.new
               = link_to "Submit Proposal", new_conference_program_proposal_path(conference.short_title), class: 'btn btn-default'
+            - if current_user && current_user.ticket_purchases.by_conference(conference).unpaid.any?
```
Comment (siddhantbajaj) [inline app/views/conferences/_conference_details.html.haml]: Just to give the option to user to pay their unpaid ticket purchases of that conference. I'll add a link in physical ticket index page as well. 🙂

Comment (Ana06) [inline app/views/conferences/_conference_details.html.haml]: Then just add the pysical tickets with the ticket name, and the unpaid ones are reachable from there, as we want a link from the pysical ticket index to the unpaid tickets. So we don't overload it. But this is not related to this PR, so in a different PR please :wink:

Comment (siddhantbajaj) [inline app/views/conferences/_conference_details.html.haml]: >Then just add the pysical tickets with the ticket name, and the unpaid ones are reachable from there.

Sorry can you please explain it a bit more?

Comment (Ana06) [inline app/views/conferences/_conference_details.html.haml]: in the conference details we could a link to the pyshical tickets index page with the text `Tickets`.

And as we want to have message with a link from the pysical tickets index page to the unpaid ones, in case that you want to reach the unpaid ones, you would have to go to tickets and then to the unpaid tickets. To avoid having many things in the conference details.

Did I explain myself better now? :wink:

**Comment by siddhantbajaj** [comment]
screen shot 2017-06-29 at 8 38 38 pm

File: app/views/physical_ticket/index.html.haml [hunk 4]
```diff
@@ -33,3 +33,11 @@
                     class: 'button btn btn-default btn-info'
       - else
         %h5 No Tickets found!
+  .row
+    .col-md-12
+      - if @unpaid_ticket_purchases.any?
+        .h3
+          Unpaid Tickets!
```
Comment (Ana06) [inline app/views/physical_ticket/index.html.haml]: What about changing it by:
`You have unpaid tickets!`

File: app/views/physical_ticket/index.html.haml [hunk 5]
```diff
@@ -33,3 +33,11 @@
...
+          %small
+            = link_to "Pay here", conference_ticket_purchases_path
+            for your unpaid tickets.
```
Comment (Ana06) [inline app/views/physical_ticket/index.html.haml]: what about removing this and in the link just adding `Pay them here`

File: app/controllers/ticket_purchases_controller.rb [hunk 6]
```diff
@@ -6,6 +6,7 @@ class TicketPurchasesController < ApplicationController
   def create
     current_user.ticket_purchases.by_conference(@conference).unpaid.destroy_all
     message = TicketPurchase.purchase(@conference, current_user, params[:tickets][0])
+
```
Comment (Ana06) [inline app/controllers/ticket_purchases_controller.rb]: why this empty line?

**Comment by Ana06** [comment]
Another PR we can merge! :+1:
